src/reader/decoder.rs

```diff
@@ -754,17 +766,39 @@ impl StreamingDecoder {
                         }
                     }
                 } else {
+                    self.ext.data.push(b);
                     self.ext.is_block_end = false;
                     goto!(ExtensionDataBlock(b as usize), emit Decoded::SubBlockFinished(self.ext.id))
                 }
             }
             ApplicationExtension => {
                 debug_assert_eq!(0, b);
                 // the parser removes sub-block lenghts, so app name and data are concatenated
-                if self.ext.data.len() >= 15 && &self.ext.data[1..13] == b"NETSCAPE2.0\x01" {
-                    let repeat = &self.ext.data[13..15];
+                if self.ext.data.len() >= EXT_NAME_NETSCAPE.len()
```

Comment:
These could be simplified with https://doc.rust-lang.org/stable/std/primitive.slice.html#method.strip_prefix